src/Servers/Kestrel/Core/src/Internal/WebTransport/WebTransportSession.cs

```diff
@@ -0,0 +1,191 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading.Channels;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3;
+using Microsoft.AspNetCore.Server.Kestrel.Core.WebTransport;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.WebTransport;
+
+internal sealed class WebTransportSession : IWebTransportSession
+{
+    private static readonly IStreamDirectionFeature _outputStreamDirectionFeature = new DefaultStreamDirectionFeature(canRead: false, canWrite: true);
+
+    private readonly CancellationTokenRegistration _connectionClosedRegistration;
+
+    // stores all created streams (pending or accepted)
+    private readonly ConcurrentDictionary<long, WebTransportStream> _openStreams = new();
+    // stores all pending streams that have not been accepted yet
+    private readonly Channel<WebTransportStream> _pendingStreams;
+
+    private readonly Http3Connection _connection;
+    private readonly Http3Stream _connectStream = default!;
+    private bool _isClosing;
+
+    private static readonly ReadOnlyMemory<byte> OutputStreamHeader = new(new byte[] {
+            0x40 /*quic variable-length integer length*/,
+            (byte)Http3StreamType.WebTransportUnidirectional,
+            0x00 /*body*/});
+
+    internal static string WebTransportProtocolValue => "webtransport";
+    internal static string SecPrefix => "sec-webtransport-http3-";
+    internal static string VersionHeaderPrefix => $"{SecPrefix}draft";
+    internal static string CurrentSuppportedVersion => $"{VersionHeaderPrefix}02";
```

Comment:
static -> const?

Reply:
This would turn the property into a variable, is this intentional? I was under the impression that we want to maximize the usage of properties if possible.